### PR TITLE
[fea] setup playwright

### LINKED ISSUE #158: Add integration tests to to CI pipeline (Playwright)
To avoid publishing a broken version of our app, we need to run integration tests before deploying to production.

I suggest implementing integration tests to validate our main workflows. This also simplifies the migration work for the next latest version.

**Tasks**:

- [x] setup playwright for our site
  - [x] run it locally
- [x] setup GitHub Actions
- [x] Implement integration tests

**References**:

- https://playwright.dev/docs/ci-intro

### PR DESCRIPTION
**Pull request by marco-souza**
# Descrição

This addresses #158 

- [x] initial setup
- [x] setup scripts
- [x] ensure CI is running
- [x] create some test cases
  - [x] rendering
  - [x] dark theme
  - [x] toggle language
  - [x] navigation links

This closes #158

### CODE REVIEW

**Review by marco-souza — COMMENTED** [review]
@frattezi @rafaellvs @melipefelgaco it's ready for review :+1:

**Comment by rafaellvs** [comment]
lgtm :ship: 

image

**Review by rafaellvs — APPROVED** [review]
lgtm :ship: 
image

**Comment by marco-souza** [comment]
Thank you, Lil Skinny! 🙌 🚀

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Playwright Tests
+on:
```
Comment (frattezi) [inline .github/workflows/playwright.yml:2]: [NIT]

I would break lines here after each command block: `on`, `jobs`, etc...

File: package.json [hunk 2]
```diff
@@ -32,7 +33,9 @@
     "react-i18next": "^11.14.0"
   },
   "devDependencies": {
+    "@playwright/test": "^1.44.1",
     "@types/lodash": "^4.14.178",
+    "@types/node": "^20.14.6",
```
Comment (frattezi) [inline package.json:38]: Let's also update engine req to Node 20? WDYT?

```
  "engines": {
    "node": ">=20"
  }
```

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Run Playwright tests
+        run: yarn playwright test
+
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: playwright-report
+          path: playwright-report/
+          retention-days: 30
```
Comment (frattezi) [inline .github/workflows/playwright.yml:42]: [NIT]

We can lower retention for this artifact right? Would it be relevant for more than like, 2 days?

File: .github/workflows/playwright.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: Playwright Tests
+on:
+  push:
+    branches: [main, master]
```
Comment (frattezi) [inline .github/workflows/playwright.yml:4]: Why master is here?

```suggestion
    branches: [main]
```

Comment (frattezi) [reply .github/workflows/playwright.yml:4]: Same below

**Review by frattezi — APPROVED** [review]
Beautiful Mr Marquinhos 💋 

Left some minors and nits in the comments, thank you for this setup 🙇🏼

File: package.json (hunk 2 above)
Comment (marco-souza) [reply package.json:38]: I tried it, but everything breaks, as we're using nextjs 12.
I think we'll need to migrate to version 14 first 🤯 

I'm planning to do that soon, so I'm creating the tests to ensure we don't break anything moving forward

File: .github/workflows/playwright.yml (hunk 4 above)
Comment (marco-souza) [reply .github/workflows/playwright.yml:4]: Nope, it was autogenerated by playwright

File: .github/workflows/playwright.yml (hunk 3 above)
Comment (marco-souza) [reply .github/workflows/playwright.yml:42]: Sure, let me change this.
It was autogenerated by playwright

Comment (frattezi) [reply .github/workflows/playwright.yml]: we can never be happy

File: package.json (hunk 2 above)
Comment (frattezi) [reply package.json:38]: Oh sorry, this is webapp 😆  You're right Mr

**Pull request merged**